src/DataFramesMeta.jl

```diff
@@ -392,10 +391,35 @@ function transform(g::GroupedDataFrame; kwargs...)
     idx1 = [1; 1 .+ idx2[1:end-1]]
     for (k, v) in kwargs
         first = v(g[1])
-        result[k] = Array{eltype(first)}(undef, size(result, 1))
-        result[idx1[1]:idx2[1], k] = first
-        for i in 2:length(g)
-            result[idx1[i]:idx2[i], k] = v(g[i])
+        if first isa AbstractVector 
+            if length(first) != size(g[1], 1)
+                throw("If a function returns a vector, the result " * 
+                      "must have the same length as the groups it " *
+                      "operates on")
+            end
+            result[k] = Array{eltype(first)}(undef, size(result, 1))
+            result[idx1[1]:idx2[1], k] = first
+            for i in 2:length(g)
+                out = v(g[i])
+                S = eltype(out)
+                T = eltype(result[k])
+                if !(S <: T)
+                    result[k] = copyto!(similar(result[k]), promote_type(S, T), result[k])
```

Comment:
You need to provide the indices of values which have been filled, as copying values which are still undefined will throw an error (not for `Int`, but e.g. for `String` columns).

Reply:
Fixed